File: server/services/commissionService.js
/**
 * Commission & Revenue Service
 * EventHub earns a configurable platform commission on confirmed vendor bookings.
 */

const DEFAULT_COMMISSION_RATE = 10.0; // 10% platform commission

const calculateCommission = (bookingAmount, customRate = null) => {
  const rate = customRate !== null ? Number(customRate) : DEFAULT_COMMISSION_RATE;
  const amount = Number(bookingAmount) || 0;
  const commissionAmount = Math.round((amount * rate) / 100);

  return {
    commissionRate: rate,
    commissionAmount,
    vendorNetPayout: amount - commissionAmount
  };
};

module.exports = {
  DEFAULT_COMMISSION_RATE,
  calculateCommission
};

File: server/services/commission_service.py
from config.db_config import COMMISSION_RATE

def calculate_commission(booking_total, custom_rate=None):
    """
    Computes EventHub platform commission from successful vendor booking.
    """
    rate = custom_rate if custom_rate is not None else COMMISSION_RATE
    commission_amount = round(booking_total * rate, 2)
    vendor_payout = round(booking_total - commission_amount, 2)
    
    return {
        'booking_total': round(booking_total, 2),
        'commission_rate': rate,
        'commission_amount': commission_amount,
        'vendor_payout': vendor_payout
    }
